crates/core/src/partials/mod.rs

```diff
@@ -1,3 +1,36 @@
+//! # Using partials
+//! To use `{% include %}` or `{% render %}` tags in a template, you first need to compile these
+//! included files as template partials.
+//!
+//! Example:
+//!
+//! ```liquid
+//! # common.liquid
+//! Number: {{ i }}
+//! ```
+//!
+//! ```rust,no_run,compile_fail
```

Comment:
Sorry, I had overlooked your comments that you had resolved.

I'm fine with `no_run` examples but I really dislike `ignore` (which is more of what this should be than `compile_fail`).

Instead, let's stop re-exporting `partials` in `liquid` and instead have a dedicated module that manually re-exports the types from `liquid_core::partials`.

What would be helpful is if we organized the PR as the following commits
- A commit to only re-export `partials` types and not the `mod` itself
- A commit that adds the documentation.

As those are the two core atomic operations for this PR.

Reply:
Got it.

There is still an issue with `ParserBuilder` being defined in in a layer above `liquid_core::partials` here, though.

So, in the rustdoc code that we include here in `crates/core/src/partials/mod.rs`, we would be showing how a user would create a partial to be included in a `liquid::ParserBuilder`, but `liquid::ParserBuilder` being a client of `liquid_core::partials` will not be in scope here to be imported in the rustdoc code.

In other words, this compiles just fine:

```rust

//! ```rust,no_run
//! use liquid_core::partials::{EagerCompiler, InMemorySource, PartialSource};

```

But the following will not:

```rust

//! ```rust,no_run
//! use liquid_core::partials::{EagerCompiler, InMemorySource, PartialSource};
//! use liquid::ParserBuilder;

```

Reply:
No, I was saying we should create `src/partials.rs` and have the example on that.

Reply:
Creating `src/partials.rs` makes sense to me. The rustdoc example that we can write in that module won't be able to include and compile this usage snippet though:

```rust
//! // Compile and render the main template, which uses the "common" partial.
//! let parser = ParserBuilder::with_stdlib().partials(partials).build().unwrap();
```

Since `liquid::ParserBuilder` depends on `partials`, not the other way around.

Reply:
Wait, nevermind, I think that'll work. Lemme prep the commits.

Reply:
Wait, nevermind, I think that'll work. I'll prep the commits.

Reply:
Done.